### PR TITLE
Customise Markdown styles and improve `docs/scss` structure

### PR DESCRIPTION
**Pull request by robbiemanson**
This customises the CSS we use to style code examples generated by Markdown files, by removing everything from the original Primer CSS except what we need. The original styles were disrupting our own code examples.

This branch also improves the structure of the `docs/scss` directory, bringing it much closer to `assets/scss` and improving the clarity of documentation-specific folders/files.

Finally, this branch also adds useful documentation for creating new components.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,7 +132,21 @@ All team members should be contributing back to this repo in order to improve it
 
 Editor preferences are available in the [editor config](https://github.com/fac/origin/blob/master/.editorconfig) for easy use in common text editors. Read more and download plugins at <http://editorconfig.org>.
 
-The process:
+### Creating a new component
+
+There are two distinct steps involved in creating a new CSS component: authoring and testing it locally as part of the docs, then copying it over to the `assets` folder and publishing out into the world via npm.
+
+1. Make sure you're up to date with the master branch (`git fetch; git pull origin master`)
```
Comment (caius) [inline README.md]: `gem install git-up; git up` :trollface:

File: README.md [hunk 2]
```diff
@@ -134,7 +132,21 @@ All team members should be contributing back to this repo in order to improve it
...
+2. Branch off master (`git checkout -b <add-amazing-component>`)
+3. Create a new `.scss` partial for your component in [`docs/assets/scss/global/components`](https://github.com/fac/origin/tree/master/docs/assets/scss/global/components) and give it a (singular, not plural) name
+4. `@import` your partial into [`docs.scss`](https://github.com/fac/origin/blob/master/docs/assets/scss/docs.scss)
+5. Create a new `.md` file in [`docs/components`](https://github.com/fac/origin/tree/master/docs/components) that matches your `.scss` partial name, and document examples of all use cases and variants of your component
+6. Test that the component renders as expected in all use cases
+7. Make a copy of your partial in [`assets/scss/global/components`](https://github.com/fac/origin/tree/master/assets/scss/global/components)
```
Comment (Haegin) [inline README.md]: :car::dash: is there any reason we don't keep these in sync automatically? (or use a symlink)?

Comment (robbiemanson) [inline README.md]: No! I didn't know we _could_. I'm all ears :)

File: README.md (hunk 1 above)
Comment (robbiemanson) [inline README.md]: Maybe in a future release :)

**Comment by caius** [comment]
Yeah looks good

**Pull request merged**